static/styles/message.css

```diff
@@ -0,0 +1,128 @@
+$first_column_width: 46px;
+$distance_of_text_elements_from_message_box_top: 8px;
+
+.message_row {
+    &.include-sender {
+        .messagebox .messagebox-content {
+            grid-template-rows: 25px repeat(3, auto);
+            padding-top: 4px;
+
+            .message_content {
+                padding: 0;
+                grid-row-start: 2;
+            }
+
+            .message_time {
+                align-self: center;
+                position: unset;
+                margin-top: 1px;
+            }
+
+            .message_edit_notice {
+                align-self: center;
+                top: 2px;
+            }
+
+            .message_controls {
+                align-self: center;
+                position: unset;
+            }
+
+            .message_sender {
+                overflow: hidden;
+                text-overflow: ellipsis;
+                grid-column: 1 / 3;
+                grid-row: 1 / 2;
+
+                > span {
+                    display: flex;
+                }
+
+                .inline_profile_picture {
+                    flex-shrink: 0;
+                    /* Let user profile take extra height without
+                       having any affect on height of the container. */
+                    position: absolute;
+                    margin-top: 6px;
+                }
+
+                .sender_name {
+                    overflow: hidden;
+                    text-overflow: ellipsis;
+                    white-space: nowrap;
+                    /* It is important to use line-height `normal` here since user's name can
+                       be in any language and `line-height: 1` doesn't work to accommodate text
+                       from start and end vertically in all languages. */
+                    line-height: normal;
+                    /* Add padding to align user name with the content */
+                    padding-left: $first_column_width;
+                    outline: none;
+                    margin-top: 3px;
+                }
+            }
+        }
+    }
+
+    .messagebox .messagebox-content {
+        display: grid;
+        align-items: start;
+        padding-left: 10px;
+        padding-right: 2px; /* For message focused border */
+        grid-template-rows: repeat(4, auto);
+        /* 60px 64px were chosen to fit message controls and time visually.
+           It is easy to adjust them in future if need be since their horizontal placement
+           will be handled by them automatically using `justify-self` property. */
+        grid-template-columns: $first_column_width auto 60px 64px;
+
+        .message_controls {
+            grid-row-start: 1;
+            grid-column-start: 3;
+            line-height: 1;
+            justify-self: end;
+            position: relative;
+            top: 6px;
```

Comment:
Is this `6px` the same as the `6px` we have as a `margin-top` for `.inline_profile_picture`? If so, extracting a `include_sender_top_margin` type constant could be very helpful to express that.

Reply:
Okay done!